client/components/domains/map-domain-step/index.jsx

```diff
@@ -20,16 +20,17 @@ var MapDomainStep = React.createClass( {
 
 	propTypes: {
 		products: React.PropTypes.object.isRequired,
+		initialQuery: React.PropTypes.string,
 		analyticsSection: React.PropTypes.string.isRequired
 	},
 
 	getInitialState: function() {
-		return { searchQuery: '' };
+		return { searchQuery: this.props.initialQuery };
```

Comment:
What do you think of removing the `searchQuery` state completely and using a query prop that is coming from the URL? When a change happens, you change the URL and the changes propagate downwards.

There's a similar example in https://github.com/Automattic/wp-calypso/blob/master/client/devdocs/main.jsx .